From 4bad38b711da2dc19c7f3e972821ec02ca3e600a Mon Sep 17 00:00:00 2001
From: Charlotte Pabst <cpabst@codeweavers.com>
Date: Thu, 29 May 2025 17:46:27 +0200
Subject: [PATCH 068/293] Revert "mf: Release pending items when sample grabber
 is stopped.".

This reverts commit 18c0dd7390b354d2e505e54cd3f3f9b08226b265.

The actual bug that broke sample delivery was in mf session code.

This commit is faulty since the Flush() method - which is called
at session Stop, but runs after sample_grabber_set_state - already
releases pending items, while processing markers immediately.
---
 dlls/mf/samplegrabber.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dlls/mf/samplegrabber.c b/dlls/mf/samplegrabber.c
index 964f4e9001e..a8c797f25ed 100644
--- a/dlls/mf/samplegrabber.c
+++ b/dlls/mf/samplegrabber.c
@@ -1191,7 +1191,6 @@ static HRESULT sample_grabber_set_state(struct sample_grabber *grabber, enum sin
                 sample_grabber_cancel_timer(grabber);
                 release_samples(grabber);
                 grabber->sample_count = MAX_SAMPLE_QUEUE_LENGTH;
-                sample_grabber_release_pending_items(grabber);
             }
 
             if (state == SINK_STATE_RUNNING && grabber->state != SINK_STATE_RUNNING)
-- 
2.43.0

